Add rsec models test

yoh2140 opened the pull request:
# WHAT
モデルテストを行う

# WHY
正常に機能するか

Commit:
update rails_helper

Commit:
create factory girl config file

Commit:
update user model. validates username maximum length 6,

Commit:
update image snd comment model. add validates :content

Commit:
create user factory,

Commit:
add user model test,

Commit:
create prototype factory,

Commit:
add prototype model test

Commit:
create image factory, and image file,

Commit:
add image model test

Commit:
create comment factory

Commit:
create comment model test

Commit:
create like factory

Commit:
add like model test

ta910 commented on app/models/comment.rb:
```diff
@@ -1,4 +1,7 @@
   belongs_to :user
+
+  validates :content, presence:true
+  
```
余分な空行消しておきましょう。

yoh2140 replied:
了解です。

ta910 commented on spec/factories/user.rb:
```diff
@@ -0,0 +1,11 @@
+  
+  factory :user do
+    password = Faker::Internet.password(6)
+    username              "aaaaaa"
```
名前もFaker使いましょう

yoh2140 replied:
了解です。

ta910 commented on spec/models/comment_spec.rb:
```diff
@@ -0,0 +1,35 @@
+
+    it "Prototypeと関連づけられている" do
+      prototype = create(:prototype)
+      comment = create(:comment, prototype: prototype)
```
コメント関連の作成は複数見られるので変数化しましょう。

yoh2140 replied:
let!でまとめました。

ta910 commented on spec/models/prototype_spec.rb:
```diff
@@ -0,0 +1,40 @@
+  describe '#validation'  do
+
+    it "正常に投稿できました。" do
+      prototype = build(:prototype)
```
protoと略すのなら統一しておきましょう。

yoh2140 replied:
了解です。

ta910 commented on spec/models/user_spec.rb:
```diff
@@ -1,7 +1,61 @@
+    end
+
+    it "7文字以上のユーザーネームは無効です。" do
+      user = build(:user, username:"aaaaaaa")
```
user関連もうまく変数化できそうですね

yoh2140 replied:
letを使い変数をまとめました。

Commit:
delete empty line,

Commit:
update user factory, username to Faker,

Commit:
update proto validates, to prototype

Commit:
update user spec to validation expect,
user validations to combine let,
let(:user) { User.new(username: nil, email: nil, password: nil)}

Commit:
update comment spec to association,
combine let,
	let!(:prototype) { create(:prototype) }
    	let!(:user) { create(:user) }
	let!(:comment) { create(:comment, prototype: prototype, user: user)}

KoichiOzaki commented:
テストコードは問題なさそうですので、rspecコマンドでテストを実行して全てsuccessで通ったら次に進んでもらって大丈夫です！
次からはrspecの実行結果を画像キャプチャして貼り付けていただけると助かります。

yoh2140 commented:
![2017-04-20 21 55 58](https://cloud.githubusercontent.com/assets/19721458/25231833/4f762f92-2614-11e7-921d-066eb02c551a.png)
次回から貼り付けるようにします。レビューありがとうございます。
